distributed/protocol/serialize.py

```diff
@@ -195,6 +202,14 @@ def deserialize(self):
         frames = decompress(self.header, self.frames)
         return deserialize(self.header, frames)
 
+    def __eq__(self, other):
+        return (isinstance(other, Serialized) and
+                other.header == self.header and
+                other.frames == self.frames)
+
+    def __ne__(self, other):
+        return not (self == other)
```

Comment:
Concerned about mutation?  I'm generally willing to assume that the user won't change values after giving them to us.  We don't really have a reasonable way of proceeding correctly if we don't make this assumption.